src/rgw/rgw_op.cc

```diff
@@ -4618,25 +4623,26 @@ void RGWPutObj::execute(optional_yield y)
 
   if (cksum_filter) {
     const auto& hdr = cksum_filter->header();
+    auto expected_ck = cksum_filter->expected(*s->info.env);
```

Comment:
For readability I think it would help to do away with the `auto` and instead spell out the actual type and comment as to what info it'll return.